templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -72,6 +72,10 @@ contents:
                 # Only leave the first 3 nameservers in /etc/resolv.conf
                 sed -i ':a $!{N; ba}; s/\(^\|\n\)nameserver/\n# nameserver/4g' /etc/resolv.tmp
                 mv -f /etc/resolv.tmp /etc/resolv.conf
+                # Workaround for bz 1929160. Reload NetworkManager to force it to
+                # re-run the lookup of the hostname now that we know we have DNS
+                # servers configured correctly in resolv.conf.
+                nmcli general reload dns-rc
```

Comment:
Strangely, `nmcli` version 1.30.6-1.fc34 doesn't show the reload command as a valid command:
```
nmcli general -h     
Usage: nmcli general { COMMAND | help }

COMMAND := { status | hostname | permissions | logging }

  status

  hostname [<hostname>]

  permissions

  logging [level <log level>] [domains <log domains>]
```

However, it's rightfully documented in the man page:
```
nmcli general {status | hostname | permissions | logging | reload} [ARGUMENTS...]
```

The man page also says that the `reload dns-rc` command "Update DNS configuration, which usually involves writing /etc/resolv.conf anew". Isn't it a problem for us? Won't it overwrite our changes to `/etc/resolv.conf`?

Reply:
Yeah, I actually delayed updating this patch because it wasn't showing up in the cli, but at some point I realized it was there anyway. :-)

The reason I used dns-rc is because https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/832 is the patch in response to this bug and it mentions using SIGUSR1 to notify NM that the DNS config had changed. In the docs, dns-rc corresponds to SIGUSR1. In my local tests it seemed to work fine, but I'm waiting on the bug reporter to try it in an environment where the bug reproduces. If it doesn't work then I may just do a general reload without parameters to reload everything.

Reply:
For reference, here is a excerpt of what `NetworkManager` man page says about `SIGUSR1`:
```
When NetworkManager is not managing DNS, the signal forces a restart of operations that depend on the DNS
configuration (like the resolution of the system hostname via reverse DNS, or the resolution of WireGuard peers);
therefore, it can be used to tell NetworkManager that the content of resolv.conf was changed externally. In the future,
further actions may be added.
```

So it sounds like something we should do, since we set `rc-manager=unmanaged` to tell NM not to write to `/etc/resolv.conf`.